### DISCUSSION TITLE
Issue with Elsa.Persistence.MongoDb.Serializers.ObjectSerializer

### DISCUSSION THREAD

**Discussion by harshad-azilen** [elsa-workflows/elsa-core#2614] [Q&A]
Hi @sfmskywalker,

How can we by pass ObjectSerializer of Elsa.Persistence.MongoDb.Serializers ? 

We are facing issue while retrieving record from mongo collection. In entity modal have property with **object type** and internally  default ObjectSerializer of Elsa.Persistence.MongoDb.Serializers called and not able to de-serialized our property. 

Any suggestion?

**Comment by sfmskywalker** [comment]
Hi @harshad-azilen 

The Elsa ObjectSerializers are registered when calling `UseMongoDbPersistence`. Right now, the only way to prevent that would be to simply not call `UseMongoDbPersistence` and instead take care of registering the necessary services yourself. Basically, you could copy the `UseMongoDbPersistence` into your own application, but leave out the call to [DatabaseRegister.RegisterMapsAndSerializers();](https://github.com/elsa-workflows/elsa-core/blob/master/src/persistence/Elsa.Persistence.MongoDb/Extensions/ServiceCollectionExtensions.cs#L54).

If you want, you can submit a PR where you can control what serializers get registered from consumer code.
